include/dav1d/picture.rs

```diff
@@ -394,3 +435,66 @@ impl From<Rav1dPicAllocator> for Dav1dPicAllocator {
         }
     }
 }
+
+impl Rav1dPicAllocator {
+    pub fn alloc_picture_data(
+        &self,
+        w: c_int,
+        h: c_int,
+        seq_hdr: Arc<DRav1d<Rav1dSequenceHeader, Dav1dSequenceHeader>>,
+        frame_hdr: Option<Arc<DRav1d<Rav1dFrameHeader, Dav1dFrameHeader>>>,
+    ) -> Rav1dResult<Rav1dPicture> {
+        let pic = Rav1dPicture {
+            p: Rav1dPictureParameters {
+                w,
+                h,
+                layout: seq_hdr.layout,
+                bpc: 8 + 2 * seq_hdr.hbd,
+            },
+            seq_hdr: Some(seq_hdr),
+            frame_hdr,
+            ..Default::default()
+        };
+        let mut pic_c = pic.to::<Dav1dPicture>();
+        // Safety: `pic_c` is a valid `Dav1dPicture` with `data`, `stride`, `allocator_data` unset.
+        let result = unsafe { (self.alloc_picture_callback)(&mut pic_c, self.cookie) };
+        result.try_to::<Rav1dResult>().unwrap()?;
+        // `data`, `stride`, and `allocator_data` are the only fields set by the allocator.
+        // Of those, only `data` and `allocator_data` are read through `r#ref`,
+        // so we need to read those directly first and allocate the `Arc`.
+        let data = pic_c.data;
+        let allocator_data = pic_c.allocator_data;
+        let mut pic = pic_c.to::<Rav1dPicture>();
+        let len = pic.p.pic_len(pic.stride);
+        // TODO fallible allocation
+        pic.data = Some(Arc::new(Rav1dPictureData {
+            // SAFETY: `MaybeUninit<u8>` should be safe for anything.
+            data: array::from_fn(|i| {
+                let ptr = data[i].unwrap().cast::<AlignedPixelChunk>();
+                let len = len[(i != 0) as usize];
+                DisjointMut::new(Rav1dPictureDataComponent { ptr, len })
```

Comment:
Is the allocated data guaranteed to be in multiples of `RAV1D_PICTURE_ALIGNMENT` and aligned to `RAV1D_PICTURE_ALIGNMENT`? And `len` is the size in bytes, not the number of chunks, right? Should we have an assert confirming that's the case? Otherwise this cast to `AlignedPixelChunk` could be incorrect.

Reply:
Oh I thought I had an alignment assert here.  Maybe I accidentally deleted it as I refactored.

Reply:
Yes, `len` is the size in bytes.  And yes, the data is required to be aligned to `RAV1D_PICTURE_ALIGNMEMT` by the C API.  I'll add the assert back.

Reply:
Added the `assert!(ptr.is_aligned())` back, deleted it accidentally.

Reply:
Also I clarified that the length is in bytes, and asserted that the length is a multiple of alignment.